wordpress-mobile/WordPress-Android
# Jetpack focus: Enable widgets on jetpack app

This PR enables home screen widgets on Jetpack app just like it is on WordPress app.  It adds
- Quick Press short cut widget
- Views this week widget

Fixes #17099, #17141, #17142


https://user-images.githubusercontent.com/990349/187333737-a348503a-927d-4f69-9008-369d107f97b8.mov


To test:

Test 1:
- Build and run Jetpack app
- Long press on Jetpack app icon in app launcher or on home screen
- Ensure it show now **Widgets** option in short cuts menu

Test 2:
- Tap on **Widgets** option
- Ensure it pops up a dialog with two widgets; a QuickPress shortcut and Views this week widget

Test 3:
- Tap and hold View this week, then drop it on home screen
- Ensure it pops up  dialog with options for Site and Color selections
- Select a Site from popup
- Select a Color; light or dark from popup
- Tap **ADD WIDGET** button which should be enable after the selections
- Ensure the widget is added to home screen

Test 4:
- You can try resizing the widget
- Widget has a wide view and a block view as shown in the video

Test 5:
- Tap on the widget
- Ensure it launches the Stats screen on insights

Test 6:
- Test Quick Press short cut too, which works just the same as on WordPress app

NOTE:
- The widget set to display **Views**, **Visitors**, **Likes** and **Comments** for this week and matches numbers on Week tab for current week.  We can update this along with other widgets if required in the next PRs.

## Regression Notes
1. Potential unintended areas of impact
Tested to make sure the widgets on WordPress app work as before and not affected by this on Jetpack app

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested WordPress app and Jetpack apps separately

3. What automated tests I added (or what prevented me from doing so)
Existing and updated unit tests

PR submission checklist:
